core/src/main/java/com/orange/ouds/core/component/InteractionValuesIndication.kt

```diff
@@ -0,0 +1,361 @@
+    val hovered: T,
+    val pressed: T,
+    val toAnimatableValue: (T) -> S,
+    val fromAnimatableValue: (S) -> T
+) {
+
+    /** The interaction state of the component this value is attached to. */
+    var interactionState: InteractionState = InteractionState.None
+        set(value) {
+            field = value
+            if (value != InteractionState.Pressed) {
+                @Suppress("KotlinConstantConditions")
+                resting = when (value) {
+                    InteractionState.None -> none
+                    InteractionState.Focused -> focused
+                    InteractionState.Hovered -> hovered
+                    InteractionState.Pressed -> error("Resting does not change when pressed.")
+                }
+            }
+        }
+
+    /**
+     * The value when animating the press interaction.
+     * It is equal to:
+     *  - the associated animatable value if the later is running
+     *  - [pressed] if the user is still pressing the component but animations are finished
+     *  - null otherwise
+     */
+    var animatingPressed: T? = null
+        set(value) {
+            field = value
+            setValue(value, resting)
+        }
+
+    /** The value when there is no press interaction. */
+    var resting: T = none
+        private set(value) {
+            field = value
+            setValue(animatingPressed, value)
+        }
+
+    /** The value. */
+    var value by mutableStateOf(none)
+
+    private fun setValue(animatingPressed: T?, resting: T) {
+        value = animatingPressed.orElse { resting }
+    }
+}
+
+/**
+ * An interaction value associated with an animatable Color.
+ */
+private class AnimatableColorInteractionValue<T>(
+    none: T,
+    focused: T,
+    hovered: T,
+    pressed: T,
+    toAnimatableColor: (T) -> Color,
+    fromAnimatableColor: (Color) -> T
+) : InteractionValue<T, Color>(none, focused, hovered, pressed, toAnimatableColor, fromAnimatableColor)
+
+/**
+ * An interaction value associated with an animatable Float.
+ */
+private class AnimatableFloatInteractionValue<T>(
+    none: T,
+    focused: T,
+    hovered: T,
+    pressed: T,
+    toAnimatableFloat: (T) -> Float,
+    fromAnimatableFloat: (Float) -> T
+) : InteractionValue<T, Float>(none, focused, hovered, pressed, toAnimatableFloat, fromAnimatableFloat)
+
+@Composable
+@JvmName("rememberAnimatableColorInteractionValue")
+private fun <T> rememberInteractionValue(
+    none: T,
+    focused: T,
+    hovered: T,
+    pressed: T,
+    interactionState: InteractionState,
+    toAnimatableColor: (T) -> Color,
+    fromAnimatableColor: (Color) -> T
+): InteractionValue<T, Color> =
+    remember(none, focused, hovered, pressed, toAnimatableColor, fromAnimatableColor) {
+        AnimatableColorInteractionValue(none, focused, hovered, pressed, toAnimatableColor, fromAnimatableColor)
+    }.apply {
+        this.interactionState = interactionState
+    }
+
+@Composable
+internal fun rememberInteractionColor(
```

Comment:
This method can be private as `rememberNullableInteractionColor` and `rememberInteractionValue` methods with all states parameters.